cargo-ext/src/cli_commands.rs

```diff
@@ -0,0 +1,48 @@
+use std::path::PathBuf;
+
+use clap::Parser;
+use clap::Subcommand;
+use midenc_session::TargetEnv;
+
+#[derive(Parser, Debug)]
+#[command(name = "cargo")]
+#[command(bin_name = "cargo")]
+pub enum CargoCli {
+    Miden(MidenArgs),
+}
+
+#[derive(Parser, Debug)]
+#[command(name = "miden")]
+#[command(bin_name = "cargo miden")]
+#[command(about = "Cargo command for developing Miden projects", long_about = None)]
+pub struct MidenArgs {
+    #[command(subcommand)]
+    pub command: Commands,
+}
+
+#[derive(Debug, Subcommand)]
+pub enum Commands {
+    /// Compile the current project to MASM
+    #[command(next_display_order(10), name = "compile")]
+    Compile {
+        /// The target environment to compile for
+        #[arg(long = "target", value_name = "TARGET", default_value_t = TargetEnv::Base, display_order(2))]
+        target: TargetEnv,
+
+        /// Tells the compiler to produce an executable Miden program from the binary target or a library from the lib target if not specified
```

Comment:
> I don't think this flag is needed, probably - we can assume `--bin` is the default build type (just like `cargo build`), and use the default binary target for the crate being compiled (using the crate name as the output file name).
> 
> We probably want to add a `--lib` flag though, to build a library target rather than a binary target. We could also flip the defaults so that `--lib` is the default and `--bin` must be specified, but either way I don't expect that multiple binary targets is likely to be a common case.
> 
> What's your feeling on it?

See https://github.com/0xPolygonMiden/compiler/pull/61#discussion_r1415552505